cpp/src/arrow/util/compression_benchmark.cc

```diff
@@ -133,6 +132,38 @@ static void ReferenceStreamingCompression(
   StreamingCompression(COMPRESSION, data, state);
 }
 
+int64_t NonStreamingCompress(Codec* codec, const std::vector<uint8_t>& data,
+                             std::vector<uint8_t>* compressed_data) {
+  const uint8_t* input = data.data();
+  int64_t input_len = data.size();
+  int64_t compressed_size = 0;
+  int64_t max_compress_len =
+      codec->MaxCompressedLen(static_cast<int64_t>(data.size()), data.data());
```

Comment:
Can we use defined variables here?

```suggestion
      codec->MaxCompressedLen(input_len, input);
```